main/FatesInterfaceMod.F90

```diff
@@ -1945,19 +1975,164 @@ subroutine UpdateFatesRMeansTStep(sites,bc_in)
+  ! vegetation in the lowest layer, along with the area fraction
+  ! of those two (which should sum to unity).
+
+  ! Arguments
+  type(ed_patch_type)   :: cpatch             ! the current patch
+  real(r8), intent(in)  :: atm_par            ! direct+diffuse PAR at canopy top [W/m2]
+  real(r8), intent(out) :: seedling_par_high  ! High intensity PAR for seedlings [W/m2]
+  real(r8), intent(out) :: par_high_frac      ! Area fraction with high intensity
+  real(r8), intent(out) :: seedling_par_low   ! Low intensity PAR for seedlings [W/m2]
+  real(r8), intent(out) :: par_low_frac       ! Area fraction with low intensity
+
+  ! Locals
+  real(r8) :: upper_par  ! The PAR intensity coming from the canopy layer above [w/m2]
+  real(r8) :: upper_area ! The area fraction of the upper canopy
+  real(r8) :: lower_par  ! The PAR intensity under the lower-most canopy [W/m2]
+  real(r8) :: lower_area ! The area fractino of the lower canopy
+  integer  :: cl         ! current canopy layer
+  integer  :: ipft       ! current PFT index
+  integer  :: iv         ! lower-most leaf layer index for the cl & pft combo
+
+
+  ! Check that there are cohorts on the current patch as this calculation is
+  ! not relevant in that case (and will result in an index error due to ncan being zero)
+  if (cpatch%countcohorts .gt. 0) then
+
+     ! Radiation intensity exiting the layer above the bottom-most
+     upper_par = 0._r8
+     upper_area = 0._r8
+     cl = max(1,cpatch%NCL_p-1)
+     do ipft = 1,numpft
+        iv = cpatch%ncan(cl,ipft)
+        upper_par = upper_par + cpatch%canopy_area_profile(cl,ipft,1)* &
+             (cpatch%parprof_pft_dir_z(cl,ipft,iv)+cpatch%parprof_pft_dif_z(cl,ipft,iv))
+        upper_area = upper_area + cpatch%canopy_area_profile(cl,ipft,1)
+     end do
+     if(upper_area>nearzero)then
+        upper_par = upper_par/upper_area
+     else
+        upper_par = 0._r8
+     end if
+  else
+     upper_par = fates_unset_r8
+  end if
+
+  ! If we do have more than one layer, then we need to figure out
+  ! the average of light on the exposed ground under the veg
+  ! Note that newly spawned patches without cohorts have a default
+  ! NCL_p of one.
+  if(cpatch%NCL_p>1) then
+
+     cl = cpatch%NCL_p
+     lower_area = 0._r8
+     lower_par  = 0._r8
+     do ipft = 1,numpft
+        iv = cpatch%ncan(cl,ipft)
+        lower_area = lower_area+cpatch%canopy_area_profile(cl,ipft,1)
+        lower_par = lower_par + &
+             cpatch%canopy_area_profile(cl,ipft,1)*&
+             (cpatch%parprof_pft_dir_z(cl,ipft,iv) + cpatch%parprof_pft_dif_z(cl,ipft,iv))
      end do
+     if(lower_area>nearzero)then
+        lower_par = lower_par / lower_area
+     else
+        lower_par = 0._r8
+     end if
 
-     return
-   end subroutine UpdateFatesRMeansTStep
-      
- end module FatesInterfaceMod
+     seedling_par_high = upper_par
+     par_high_frac     = (1._r8-lower_area)
+     seedling_par_low  = lower_par/lower_area
```

Comment:
I think this should be changed to `seedling_par_low = lower_par` given the above calculation, correct?  Otherwise, you could get a divide by zero case here if `lower_area` is exactly zero.